--- a/include/quickjspp.hpp
+++ b/include/quickjspp.hpp
@@ -406,10 +406,14 @@ namespace qjs {
             case JS_TAG_BOOL:
                 return is_boolean<T>::value || std::is_integral_v<T> || std::is_floating_point_v<T>;
 
+#ifdef JS_TAG_BIG_DECIMAL
             case JS_TAG_BIG_DECIMAL:
                 [[fallthrough]];
+#endif
+#ifdef JS_TAG_BIG_FLOAT
             case JS_TAG_BIG_FLOAT:
                 [[fallthrough]];
+#endif
             case JS_TAG_FLOAT64:
             default: // >JS_TAG_FLOAT64 (JS_NAN_BOXING)
                 return is_double<T>::value || std::is_floating_point_v<T>;
@@ -474,10 +478,14 @@ namespace qjs {
             case JS_TAG_EXCEPTION:
                 break;
 
+#ifdef JS_TAG_BIG_DECIMAL
             case JS_TAG_BIG_DECIMAL:
                 [[fallthrough]];
+#endif
+#ifdef JS_TAG_BIG_FLOAT
             case JS_TAG_BIG_FLOAT:
                 [[fallthrough]];
+#endif
 
             case JS_TAG_FLOAT64:
                 [[fallthrough]];
